src/Tasks/LockCheck.cs

```diff
@@ -209,11 +211,16 @@ internal static IEnumerable<ProcessInfo> GetLockingProcessInfos(params string[]
             }
 
             const int maxRetries = 6;
+            uint handle;
+            int res;
 
-            // See http://blogs.msdn.com/b/oldnewthing/archive/2012/02/17/10268840.aspx.
-            var key = new StringBuilder(new string('\0', CCH_RM_SESSION_KEY + 1));
+            unsafe
+            {
+                // See http://blogs.msdn.com/b/oldnewthing/archive/2012/02/17/10268840.aspx.
+                char* key = stackalloc char[CCH_RM_SESSION_KEY + 1];
```

Comment:
nit: I would zero-init the array for the peace of mind. `stackalloc` seems to be finnicky:
https://stackoverflow.com/questions/8679052/initialization-of-memory-allocated-with-stackalloc

This is used on error paths so perf is not a huge concern.

Reply:
Reading the second answer makes me think we don't need to:
https://stackoverflow.com/a/53173980/908889

This is because we are using the P/Invoke directly in a method which has lots of local variables. Meaning the buffer will be zero initialized.

Being explicit is probably for the best though.

Reply:
We don't need to _currently_, but if it's officially undefined, that could change without warning. Relying on it means that, in the (admittedly somewhat unlikely) event it changes, we'd suddenly have broken code and no idea why.

Reply:
@stephentoub do we expect more formal definition of expectations here? It seems a little odd to not define whether we zero or not -- might as well not zero at all if we can't 100% rely on it. I could imagine zeroing iff SkipLocalsInit is applied, for example,

Reply:
[SkipLocalsInit] was added primarily because it would have been a massive breaking change for the C# compiler to stop outputting initlocals in places it already was.  There are one or two corner cases today where the C# compiler doesn't output initlocals, and those are essentially bugs in the compiler, whether or not there's any plan or intention to fix them.  We can rely on initlocals being in place tomorrow everywhere it's in place today.

Reply:
Thank you @stephentoub! What is the easiest way to check is `[SkipLocalsInit]` is applied?

Reply:
> What is the easiest way to check is [SkipLocalsInit] is applied?

Considering a search https://github.com/dotnet/msbuild/search?q=SkipLocalsInit through the whole repo yields 0 hits, it's almost certainly not.  You'd see it applied to something, like the particular method in question, or the type containing it, or the whole module, e.g. `[module: SkipLocalsInit]` in some file.

Reply:
Thank you. 

@ladipro / @Forgind are you both happy for me to remove the zero initializing code for the buffer since `SkipLocalsInit` isn't used within MSBuild?

Reply:
@elachlan my question was whether we could rely on zeroing being taken care of if SkipLocalsInit was NOT applied (if it WAS applied, I would certainly expect zeroing to be skipped). It's apparently not being applied. So the question remains do you need to do zeroing.

> We can rely on initlocals being in place tomorrow everywhere it's in place today.

I'm still unclear (sorry) whether this means that in the absence of SkipLocalsEmit the stackalloc buffer will be dependably zeroed. @jaredpar what can we assume about this?

Reply:
> So the question remains do you need to do zeroing.

I don't see why zero'ing would be needed here.  This is an out parameter, so the Win32 function being called doesn't care about the contents, and this buffer is ignored by the rest of the method.

Reply:
@stephentoub I think the answer for `RmStartSession` in particular is documented here:
https://devblogs.microsoft.com/oldnewthing/20120217-00/?p=8283

> The Rm­­StartSession function doesn’t properly null-terminate the session key, even though the function is documented as returning a null-terminated string. To work around this bug, we pre-fill the buffer with null characters so that whatever ends gets written will have a null terminator (namely, one of the null characters we placed ahead of time).

Reply:
But `key` is unused after the call on L221 (unless I'm missing where it's used?).  It doesn't matter what data is in it if the code never reads anything from it.  Am I missing something?

Reply:
You are correct. I think in this particular case it isn't needed. The documentation on the function says 

> A secondary installer can join an existing Restart Manager session by calling the RmJoinSession function with the session handle and **session key** returned from the RmStartSession function call of the primary installer.
https://docs.microsoft.com/en-au/windows/win32/api/restartmanager/nf-restartmanager-rmstartsession

Since we don't use that, we won't need the key as far as I can tell.